sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/metadata/MetadataContextFilter.java

```diff
@@ -0,0 +1,64 @@
+package org.sdase.commons.server.dropwizard.metadata;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.MultivaluedMap;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * A filter that creates the {@link MetadataContext} for processing a request. The {@link
+ * MetadataContext} is based on the header fields of the incoming request and includes the
+ * configured {@linkplain #MetadataContextFilter(Set) fields}.
+ */
+@Priority(Priorities.USER)
+public class MetadataContextFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+  private final Set<String> metadataFields;
+
+  public MetadataContextFilter(Set<String> metadataFields) {
+    this.metadataFields = metadataFields;
+  }
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    var metadataContext = buildContext(requestContext.getHeaders());
```

Comment:
Better safe than sorry: Should we also add `MetadataContextHolder.clear();` here?

Reply:
Hmmm, either we jump out with an exception here and the context is not used anymore or `set` in the next line will overwrite it. I'm not sure how expensive thread local operations are.